tests/acct_mgt_test.py

```diff
@@ -0,0 +1,353 @@
+       openssl pkcs12 -export -in ./acct-mgt-2.crt -inkey ./acct-mgt-2.key -out acct-mgt-2.p12
+
+    2) call curl with
+       curl -v -kv -E ./client.p12:password http://url...
+
+    3) auth_opts can be of the following:
+
+          auth_opts = ["-E","./client_cert/acct-mgt-2.crt", "-key", "./client_cert/acct-mgt-2.key"]
+          auth_opts = ["-cert", r"acct-mgt-2",]
+
+ Initial test to confirm that something is working
+    curl -kv https://acct-mgt.apps.cnv.massopen.cloud/projects/acct-mgt
+    curl -u <user>:<password> -kv https://acct-mgt.apps.cnv.massopen.cloud/projects/acct-mgt
+
+  -- testing with no authentication:
+     python3 -m pytest amt_test*.py --amurl http://acct-mgt..massopen.cloud
+
+  -- testing with basic authentication
+     python3 -m pytest amt_test*.py --amurl https://acct-mgt.massopen.cloud --basic "user:pass"
+"""
+
+import subprocess
+import re
+import time
+import json
+import pprint
+
+
+def get_pod_status(project, pod_name):
+    """This gets the specified pod status from openshift"""
+    result = subprocess.run(
+        ["oc", "-n", project, "-o", "json", "get", "pod", pod_name],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=False,
+    )
+    if result.returncode == 0:
+        result_json = json.loads(result.stdout.decode("utf-8"))
+        print(result_json["status"]["phase"])
+        return result_json["status"]["phase"]
+    print("None")
+    return None
+
+
+#
+def wait_while(project, pod_name, statuses, time_out=300):
+    """
+    This basically waits while a pod is in a set of statuses and possibly times out
+    It is used to wait for openshift to do something with the pod, as in, deleting
+    or starting a pod.
+
+    pass in the following parameter:
+       project:  the namespace of the pod
+       pod_name: the name of the pod
+       statuses: array of statuses
+
+
+    """
+    time_left = time_out
+    time_interval = 5
+    time.sleep(time_interval)
+    status = get_pod_status(project, pod_name)
+    while status in statuses and time_left > 0:
+        time.sleep(time_interval)
+        time_left = time_left - time_interval
+        status = get_pod_status(project, pod_name)
+
+    if status in statuses:
+        return False
+    return True
+
+
+def wait_until_done(oc_cmd, finished_pattern, time_out=30, decrement=5):
+    """
+    This wait while an oc command is running, looking for a pattern that
+    indicates it is finished.
+    """
+    pattern = re.compile(finished_pattern)
+    done = False
+    oc_array = oc_cmd.split(" ")
+    matched_line = ""
+    while time_out > 0 and not done:
+        time.sleep(5)
+        time_out = time_out - decrement
+        result = subprocess.run(
+            oc_array,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+        line_list = result.stdout.decode("utf-8").split("\n")
+        for line in line_list:
```

Comment:
I would prefer to see these tests looking at explicit attributes in the JSON output, rather than using regular expressions to parse the human readable output (because it's easy to get a regex wrong, but much harder to make a mistake when you say that `status.phase == "Active"` must be `True`.

Or if you prefer, make use of the `custom-columns` output format:

```
$ oc get project test-001 -o custom-columns=phase:status.phase --no-headers
Active
```

By reducing the output to just the field you want to check you may be able to use simple string matches rather than regular expressions.

Reply:
compare results was used for all comparisons, now it is mainly the ones from curl checking the results returned by the micro-service.  I have changed similar checks when you previously suggested this which is why this is mostly used for what curl passes back.

I was trying to keep this PR focused on just creating the modules.